lewis/adapters/epics.py

```diff
@@ -321,7 +321,7 @@ def _create_getter(self, func, *targets):
             raise RuntimeError(
                 'The function \'{}\' does not look like a getter function. A valid getter '
                 'function has no arguments that do not have a default. The self-argument of '
-                'methods does not count towards that number.')
+                'methods does not count towards that number.'.format(final_callable.__name__))
```

Comment:
Potentially re-introduces the same issue as was fixed in #229: 
`__call__`able `class`es and `functools.partial`s have no `__name__`.

Except `if not self._function_has_n_args(final_callable, 0):` right above this would already throw an exception, since it tries to use `inspect.getargspec` on `final_callable`, which also doesn't work with classes and partials.

Do we ever expect / want to support the `func` parameter of `_create_getter` or `_create_setter` to be a callable class or partial? 

If not, it should be fine to leave everything as-is. 
If yes, we need to fix `_function_has_n_args()` and replace `.__name__` with something that will also work for those cases (perhaps a special `utils.get_callable_name()` function, since this already came up in #229).